Upload over external serial to usb adapter (I tried the same with the new CDC functionality of the core):

```
offline> connect
No port specified - connecting to /dev/ttyUSB0 at 115200bps
Printer is now online
ttyUSB0 23°> M21
SENDING:M21
SD card ok     
ttyUSB0 23°> M20
SENDING:M20
Begin file list
End file list  
ttyUSB0 23°> load test.gcode
Loading file: test.gcode
Loaded test.gcode, 154073 lines.
Estimated duration: 321 layers, 1:43:12
ttyUSB0 23°> M28 upload.txt
SENDING:M28 UPLOAD.TXT
Now fresh file: UPLOAD.TXT
Writing to file: UPLOAD.TXT
Error:No Checksum with line number, Last Line: 0
ttyUSB0 23°> [ERROR] Error:No Checksum with line number, Last Line: 0

Resend: 1      
ttyUSB0 23°> print
Printing test.gcode
You can monitor the print with the monitor command.
ttyUSB0 23° 0%> Print started at: 13:44:42
Error:error writing to file
ttyUSB0 23° 0%> [ERROR] Error:error writing to file

Error:error writing to file
ttyUSB0 23° 0%> [ERROR] Error:error writing to file

Error:error writing to file
ttyUSB0 23° 1%> [ERROR] Error:error writing to file

Error:error writing to file
ttyUSB0 23° 1%> [ERROR] Error:error writing to file

Error:No Checksum with line number, Last Line: 2628
ttyUSB0 23° 1%> [ERROR] Error:No Checksum with line number, Last Line: 2628
....
```

Would be great if you could test it on your side. Maybe / Most likely this problem is unrelated to this PR, because the PR for the writing is doing the right thing.

--- a/Marlin/src/HAL/HAL_STM32/HAL_spi_STM32.cpp
+++ b/Marlin/src/HAL/HAL_STM32/HAL_spi_STM32.cpp
@@ -116,6 +116,7 @@ uint8_t spiRec(void) {
  */
 void spiRead(uint8_t* buf, uint16_t nbyte) {
   if (nbyte == 0) return;
+  memset(buf, 0xff, nbyte);
   SPI.beginTransaction(spiConfig);
   SPI.transfer(buf, nbyte);
   SPI.endTransaction();

--- a/Marlin/src/HAL/HAL_STM32/HAL_spi_STM32.cpp
+++ b/Marlin/src/HAL/HAL_STM32/HAL_spi_STM32.cpp
@@ -116,7 +116,7 @@ uint8_t spiRec(void) {
  */
 void spiRead(uint8_t* buf, uint16_t nbyte) {
   if (nbyte == 0) return;
-  memset(buf, 0xff, nbyte);
+  memset(buf, 0xFF, nbyte);
   SPI.beginTransaction(spiConfig);
   SPI.transfer(buf, nbyte);
   SPI.endTransaction();

